src/pages/reference/providers/terraform.mdx

```diff
@@ -0,0 +1,75 @@
+
+<Note>
+  Terraform providers are currently in preview, they're not recommended for production deployments. We recommend using the [Standard Provider](/reference/providers) for production deployments until the Terraform providers are stable.
+</Note>
+
+## Terraform Providers
+
+The Nitric Terraform providers allow you to deploy Nitric applications using Terraform. Currently, a Terraform provider is in Preview for AWS, with support for other cloud providers coming soon. While the standard Nitric Providers deploy your application directly to the cloud, the Terraform providers act more like a Transpiler for the cloud infrastructure requirements, allowing Nitric to generate a Terraform stack using [CDKTF](https://developer.hashicorp.com/terraform/cdktf).
+
+The main benefits of using the Nitric Terraform provider are:
+
+- **Infrastructure from Code**: Define your application using Nitric and automatically generate the necessary Terraform code.
+- **Terraform Ecosystem**: Leverage the existing Terraform ecosystem for managing your cloud infrastructure.
+- **Customization**: Customize your cloud infrastructure using Terraform modules and resources.
+
+You're also able to [create your own Terraform provider](/reference/terraform/custom-providers) for Nitric, this is the best option when you have existing Terraform Modules and expertise you want to reuse to deploy Nitric app.
+
+<Note>
+  If you're new to Terraform and/or Nitric, we recommend starting with a
+  [standard Nitric provider](/reference/provider). The Terraform provider is
+  more advanced and requires a foundational understanding of Terraform.
+</Note>
+
+## Enabling Nitric Terraform Providers
+
+The Nitric Terraform providers are currently in preview, to enable them you'll need to enable `beta-providers` in your Nitric project. You can do this by adding the following to your project's `nitric.yaml` file:
+
+```yaml
+preview:
+  - beta-providers
+```
+
+<Note>
+  The `beta-providers` flag enables all non-stable providers, including the
+  Terraform providers.
+</Note>
+
+## Getting Started
+
+To get started with the Nitric Terraform provider, you'll need to install the Nitric CLI and the Terraform CLI. You can find the installation instructions for both in the [Nitric CLI](/reference/cli) and [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli) documentation.
+
+Once you have the Nitric CLI and Terraform CLI installed, you can create a new Nitric project as usual, if you're new to Nitric, check out the [Getting Started](/getting-started) guide.
+
+To use the Nitric Terraform provider, you'll need to create a new nitric stack that uses the Terraform provider. You can do this by running the following command:
+
+```bash
+nitric stack new
+```
+
+This will create a new Nitric stack and prompt you to select a provider. Select the Terraform provider of your choice during this process.
```

Comment:
Do we show the beta providers in the stack templates if its enabled in the project?

Reply:
I have added a change in this CLI PR that will show if they select one. https://github.com/nitrictech/cli/pull/738/commits/380ac94083a8ee11d8538fc1c699b7cda89119cb